internal/device/energy_zone.go

```diff
@@ -0,0 +1,154 @@
+// SPDX-FileCopyrightText: 2025 The Kepler Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package device
+
+import (
+	"fmt"
+	"math"
+	"sync"
+)
+
+type Zone = string
+
+const (
+	ZonePackage Zone = "package"
+	ZoneCore    Zone = "core"
+	ZoneDRAM    Zone = "dram"
+	ZoneUncore  Zone = "uncore"
+	ZonePSys    Zone = "psys"
+	ZonePP0     Zone = "pp0" // Power Plane 0 - processor cores
+	ZonePP1     Zone = "pp1" // Power Plane 1 - uncore (e.g., integrated GPU)
+)
+
+// zoneKey uniquely identifies a zone by name and index
+type zoneKey struct {
+	name  string
+	index int
+}
+
+// AggregatedZone implements EnergyZone interface by aggregating multiple zones
+// of the same type (e.g., multiple package zones in multi-socket systems).
+// It handles energy counter wrapping for each individual zone and provides
+// a single consolidated energy reading.
+type AggregatedZone struct {
+	name          string
+	index         int
+	zones         []EnergyZone
+	lastReadings  map[zoneKey]Energy
+	currentEnergy Energy // Aggregated energy counter
+	maxEnergy     Energy // Cached sum of all zone MaxEnergy values
+	mu            sync.RWMutex
+}
```

Comment:
Can we add a method `Label` in interface, for which `AggregatedZone` will return just name, and sysFsRaplZone will return "{Name}-{Index}" ?
And we use `Label` in power collector?

Reply:
Since we are replacing multiple zones with a single zone, and won't be exposing the sub zones I don't think having a label helps in this case.

Reply:
I will mark this as resolved, please re-open if I didn't understand this correctly.